app/backend/src/couchers/interceptors.py

```diff
@@ -123,35 +127,45 @@ def _store_log(self, method, status_code, duration, user_id, request, response,
                     user_id=user_id,
                     request=req_bytes,
                     response=res_bytes,
+                    details=grpc_details,
                     traceback=traceback,
                 )
             )
-        logger.debug(f"{user_id=}, {method=}, {duration=} ms")
+        logger.debug(f"{user_id=}, {method=}, {grpc_details=}, {duration=} ms")
 
     def intercept_service(self, continuation, handler_call_details):
         handler = continuation(handler_call_details)
         prev_func = handler.unary_unary
         method = handler_call_details.method
 
         def tracing_function(request, context):
+            user_id = getattr(context, "user_id", None)
+            start = perf_counter_ns()
+            res, code, traceback, exception_type, exception, details = None, None, None, None, None, None
+
             try:
-                start = perf_counter_ns()
                 res = prev_func(request, context)
-                finished = perf_counter_ns()
-                duration = (finished - start) / 1e6  # ms
-                user_id = getattr(context, "user_id", None)
-                self._store_log(method, None, duration, user_id, request, res, None)
             except Exception as e:
+                with context._state.condition:
+                    code_object = context._state.code
+                    if code_object != None:
+                        code = code_object.name
+                    binary_details = context._state.details
+                    if binary_details != None:
+                        details = binary_details.decode('utf-8')
+                traceback = "".join(format_exception(exception_type, e, e.__traceback__))
+                exception = e
+                exception_type = type(e)
+            finally:
                 finished = perf_counter_ns()
```

Comment:
I had the perf counter end asap to try to only time the function, not exception handling, etc